site/sfguides/src/kepler_gl_maps_inside_snowflake/kepler_gl_maps_inside_snowflake.md

```diff
@@ -0,0 +1,225 @@
+Duration: 5
+
+Let’s construct a map that visualizes UK highways with color-coded density of nearby EV charging stations. We’ll use SQL queries to filter and aggregate data from Overture Maps, focusing on major roads and charging infrastructure.
+
+![Build Map in Dekart](assets/kepler_gl_snowflake_11.png)
+
+### Define the UK Boundary
+
+To create a map specific to the UK, we need to define the boundaries of the UK region. This can be done using the **DIVISION_AREA** table from Overture Maps, which contains geographic boundaries for different regions.
+
+![Define UK Boundary](assets/kepler_gl_snowflake_12.png)
+
+```sql
+-- Define the UK boundary as a geographic region
+SELECT ST_ASWKT(GEOMETRY) AS GEOMETRY_WKT
+FROM OVERTURE_MAPS__DIVISIONS.CARTO.DIVISION_AREA
+WHERE COUNTRY = 'GB' AND SUBTYPE = 'country';
+```
+
+Explanation:
+- `ST_ASWKT(GEOMETRY)`: Converts the geometric shape of the UK boundary into a format compatible with Kepler.gl.
+- This query filters by `COUNTRY = 'GB'` to select only boundaries relevant to the United Kingdom and limits results to the outer boundary using `SUBTYPE = 'country'`.
+
+### Select Major Road Segments
+
+To focus on major roads, such as highways and trunk roads, we’ll query the **SEGMENT** table from Overture Maps. This will limit the map to primary roadways that are most relevant to our EV charging station analysis.
+
+![Select Major Road Segments](assets/kepler_gl_snowflake_13.png)
+
+```sql
+SELECT ST_ASWKT(s.GEOMETRY) AS GEOMETRY_WKT, s.NAMES, s.ID
+FROM OVERTURE_MAPS__TRANSPORTATION.CARTO.SEGMENT s, uk_boundary ub
+WHERE ST_INTERSECTS(ub.GEOMETRY, s.GEOMETRY) AND s.CLASS IN ('motorway', 'trunk');
+```
+
+Explanation:
+- This query uses `ST_INTERSECTS` to find road segments that lie within the defined UK boundary.
+- We filter to include only major roads, such as `motorway` and `trunk` roads, for a focused map of primary road infrastructure.
+
+### Select EV Charging Stations
+
+Next, we’ll locate EV charging stations within the UK boundary. Using the **PLACES** table, we can filter for points of interest categorized as charging stations.
+
+![Select EV Charging Stations](assets/kepler_gl_snowflake_14.png)
+
+```sql
+SELECT ST_ASWKT(p.GEOMETRY) GEOMETRY
+FROM OVERTURE_MAPS__PLACES.CARTO.PLACE p, uk_boundary ub
+WHERE ST_CONTAINS(ub.GEOMETRY, p.GEOMETRY) AND p.CATEGORIES::TEXT ILIKE '%charging%';
+```
+
+Explanation:
+- `ST_CONTAINS`: Ensures that only charging stations within the UK boundary are selected.
+- `ILIKE '%charging%'`: Filters points of interest to include only those that contain “charging” in their category, ensuring relevance to EV infrastructure.
+
+### Calculate Charging Station Density Near Roads
+
+We’ll calculate the density of charging stations within a 50-kilometer radius of each road segment. This helps identify areas along highways with higher or lower access to charging facilities.
+
+![Calculate Charging Station Density Near Roads](assets/kepler_gl_snowflake_15.png)
```

Comment:
Does customer need to apply some manipulations with UI to get this visualisation? E.g set stroke colour based on the 'num_charging_stations' field and amend the colour map

Reply:
It's covered in Style the Map in Dekart